data_model/src/block.rs

```diff
@@ -271,16 +307,26 @@ mod candidate {
             self.validate_signatures()?;
             self.validate_header()?;
 
-            if self.payload.transactions.is_empty() {
-                return Err("Block is empty");
-            }
-
             Ok(SignedBlockV1 {
-                payload: self.payload,
                 signatures: self.signatures,
+                payload: self.payload,
             })
         }
 
+        fn validate_signatures(&self) -> Result<(), &'static str> {
+            self.signatures
+                .iter()
+                .map(|signature| signature.0)
+                .try_fold(BTreeSet::new(), |mut acc, elem| {
+                    if !acc.insert(elem) {
+                        return Err("Duplicate signature in block");
+                    }
+
+                    Ok(acc)
+                })?;
```

Comment:
The same as

Reply:
here it's different because we should compare signature every to every other